media/filters/chunk_demuxer.h

```diff
@@ -597,6 +619,10 @@ class MEDIA_EXPORT ChunkDemuxer : public Demuxer {
   std::vector<std::unique_ptr<ChunkDemuxerStream>> removed_streams_;
 
   std::map<MediaTrack::Id, ChunkDemuxerStream*> track_id_to_demux_stream_map_;
+
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+  std::map<std::string, std::string> id_to_mime_map_;
```

Comment:
How large is this container going to be? Consider `base::flat_map` or `base::small_map`: 
https://chromium.googlesource.com/chromium/src/+/main/base/containers/README.md#Map-and-set-selection-Usage-advice

Reply:
I'm not sure, but as there's a new id for each SourceBuffer, and the size of the map depends on how fast the StarboardRenderer is reading from the ChunkDemuxer, I think it's fair to assume it can grow large. I can't really tell if `base::flat_map` would be to our advantage.

Reply:
If we don't have any insight on the container behaviour/size, it's OK to leave a vanilla std::map. There's a lot of special containers in https://chromium.googlesource.com/chromium/src/+/main/base/containers/README.md - maybe one could apply here. But as-is is fine as well.